docs/manual/working/javaGuide/main/templates/JavaTemplatesDependencyInjection.md

```diff
@@ -0,0 +1,41 @@
+# Dependency Injection with Templates
+
+Twirl templates can now be created with a constructor annotation using `@this`.  The constructor annotation means that Twirl templates can be injected into templates directly and can manage their own dependencies, rather than the controller having to manage dependencies not only for itself, but also for the templates it has to render.
+
+As an example, suppose a template has a dependency on a component `TemplateRenderingComponent`, which is not used by the controller.  
+
+First, add the `@Inject` annotation to Twirl in `build.sbt`:
+
+```scala
+TwirlKeys.constructorAnnotations += "@javax.inject.Inject()"
```

Comment:
@mkurz The problem I think is that Twirl can be used OUTSIDE of Play, and if you see the twirl docs, they don't assume Play is the target (outside of this PR).